File: javascript/src/stacks/0682-baseball-game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
const calPoints = function (operations) {
  let stack = [];
  let sum = 0;

  for (const o of operations) {
    if (o === "+") {
      const val = stack[stack.length - 1] + stack[stack.length - 2];
      stack.push(val);
      sum += val;
      continue;
    }

    if (o === "D") {
      const val = stack[stack.length - 1] * 2;
      stack.push(val);
      sum += val;
      continue;
    }

    if (o === "C") {
      sum -= stack.pop();
      continue;
    }

    const val = parseInt(o);
    stack.push(val);
    sum += val;
  }

  return sum;
};

module.exports = calPoints;

File: golang/stacks/0682-baseball-game.go
package stacks

import "strconv"

func calPoints(operations []string) int {
	sum := 0
	stack := []int{}

	for _, o := range operations {
		n := len(stack)

		if o == "+" {
			v := stack[n-1] + stack[n-2]
			sum += v
			stack = append(stack, v)
			continue
		}

		if o == "C" {
			sum -= stack[n-1]
			stack = stack[:n-1]
			continue
		}

		if o == "D" {
			v := stack[n-1] * 2
			sum += v
			stack = append(stack, v)
			continue
		}

		num, _ := strconv.Atoi(o)
		sum += num
		stack = append(stack, num)

	}

	return sum
}
